### PR TITLE
Simplify workspace allocation in execution engine

### PR DESCRIPTION
**Pull request by lptr**

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test this please.

**Comment by lptr** [comment]
@wolfs I'm planning to merge this as soon as the tests pass. PTAL at your earliest convenience, and we can follow up in a new PR if needed.

File: subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/cache/KotlinDslWorkspaceProvider.kt [hunk 1]
```diff
@@ -31,11 +34,19 @@ class KotlinDslWorkspaceProvider(
...
+        cacheRepository
+            .cache(cacheScopeMapping.getBaseDirectory(null, "kotlin-dsl", VersionStrategy.CachePerVersion))
+            .withDisplayName("kotlin-dsl"),
+        fileAccessTimeJournal,
+        inMemoryCacheDecoratorFactory,
+        stringInterner
+    )
+
+    override fun <T : Any> withWorkspace(path: String, action: WorkspaceProvider.WorkspaceAction<T>): T =
+        delegate.withWorkspace("$accessorsWorkspacePrefix/$path", action)
```
Comment (wolfs) [inline subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/cache/KotlinDslWorkspaceProvider.kt:49]: Isn't the cleanup depth wrong here, given that we introduce a directory in between?

**Review by wolfs — COMMENTED** [review]
I left one comment.

**Pull request merged**